File: app/src/main/java/fr/free/nrw/commons/wikidata/WikidataEditListenerImpl.java
package fr.free.nrw.commons.wikidata;

/**
 * Listener for wikidata edits
 */
public class WikidataEditListenerImpl extends WikidataEditListener {

  public WikidataEditListenerImpl() {
  }

  /**
   * Fired when wikidata P18 edit is successful. If there's an active listener, then it is fired
   */
  @Override
  public void onSuccessfulWikidataEdit() {
    if (wikidataP18EditListener != null) {
      wikidataP18EditListener.onWikidataEditSuccessful();
    }
  }
}

File: app/src/main/java/fr/free/nrw/commons/wikidata/WikidataEditListenerImpl.kt
package fr.free.nrw.commons.wikidata

/**
 * Listener for wikidata edits
 */
class WikidataEditListenerImpl : WikidataEditListener() {
    /**
     * Fired when wikidata P18 edit is successful. If there's an active listener, then it is fired
     */
    override fun onSuccessfulWikidataEdit() {
        authenticationStateListener?.onWikidataEditSuccessful()
    }
}
